q2_cutadapt/tests/test_trim.py

```diff
@@ -0,0 +1,194 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2017, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import os
+import unittest
+
+from q2_cutadapt._trim import _build_trim_command
+from q2_types.per_sample_sequences import (
+    CasavaOneEightSingleLanePerSampleDirFmt,
+    SingleLanePerSampleSingleEndFastqDirFmt,
+    SingleLanePerSamplePairedEndFastqDirFmt,
+    FastqGzFormat,
+)
+from qiime2 import Artifact
+from qiime2.util import redirected_stdio
+from qiime2.plugin.testing import TestPluginBase
+
+
+class TestTrimSingle(TestPluginBase):
+    package = 'q2_cutadapt.tests'
+
+    # This test is really just to make sure that the command runs - the
+    # detailed tests in the Util Tests below ensure the commands are crafted
+    # appropriately.
+    def test_typical(self):
+        demuxed = Artifact.import_data('SampleData[SequencesWithQuality]',
+                                       self.get_data_path('single-end'))
+        with redirected_stdio(stdout=os.devnull):
+            self.plugin.methods['trim_single'](demuxed)
+        # TODO: test results
```

Comment:
I think you're adding these tests now? They can be pretty minimal - remember, your goal isn't to test the inner-workings of `cutadapt`, just that you're calling it correctly, that it's doing something when you call it, and that the output files are what you expect. For example, you can test that one or more reads are different in a couple of the fastq files - that will ensure that your looping over files is working as expected, and that your command is creating output files that are different from the input files. (When testing the paired-end command, you'll also want to make sure that the output forward reads correspond to the input forward reads, and the output reverse reads correspond to the input reverse reads - in other words, the order didn't get mixed up.)